### ISSUE TITLE
Should vehicles be able to cross road when spawning

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#505]
In EPIC (todo: add issue ref once added), the discussion focuses on vehicles crossing road to reach final destination:

* Parking in a building
* Their despawn point (eg. hearse returns to cemetery and despawns)

What I have not yet given sufficient thought to is what happens when vehicles first spawn from a building? Should they be allowed to cross the road to quickly head in the right direction - and how would that be achieved?

One brief ponderance:

* Pathfinder routes vehicle from source to destination
* If it sees the route cross the source segment a second time (ie. vehicle had to go turn somewhere and then ends up passing its source building but on other side of road) the prior route could be ditched (only if more than X segments?) if the vehicle can feasibly cross the road immediately after spawning.

We would likely need similar limits to the final destination crossing features, for example filter by vehicle type / state, and also prevent near junctions.

**Comment by kvakvs** [comment]
Comparing the path for segment ID repeated is not reliable. I'd allow the vehicle crossing the road unless there's a physical median. Again i do not know all traffic rules in the world, but those I do know allow vehicle crossing the road if leaving an adjacent land lot.

**Comment by aubergine10** [comment]
I think both for source and destination we need to factor in not just the road, but also traffic congestion/density in the decision making process. If oncoming traffic is heavy, trying to cross road would likely be bad idea.

As for traffic rules worldwide, I assume authoritarian countries like USA, China, etc., would punish drivers for doing such things.

**Comment by kvakvs** [comment]
This kind of raises a question.
A switchable configuration for road rules in a country, with all the permitted and forbidden actions, which can be used by the pathfinder.

**Comment by aubergine10** [comment]
Yes, that's been discussed several times. It would also include things like choice of road sign style, turn on red, etc.

I think a first step towards that would be to at least have flags in the global config xml that decide how all these things work. Then, later, we can think about having regional profiles which basically just stored sets of flags.

I'm not sure if there is already an issue to track that discussion, if not please create an epic as it is a recurrent theme of discussion and there are already some issues that relate to it.

**Comment by kvakvs** [comment]
I don't think it is worth its own ticket yet, but at least worth mentioning in the routing ticket.

**Comment by aubergine10** [comment]
I will repurpose that routing epic to be about the 'regional settings'.

All the other stuff is going in separate epics that are more focussed on related groups of issues.
